From 960ef523cfe420ae8fdd801f1d12be0fb7ed8719 Mon Sep 17 00:00:00 2001
From: Alex Tran <alex.t.tran@gmail.com>
Date: Mon, 1 Sep 2025 11:40:06 -0700
Subject: [PATCH] ASoC: codecs: tlv320dac33: Add default value for burst_bclkdiv
Git-commit: 960ef523cfe420ae8fdd801f1d12be0fb7ed8719
Patch-mainline: v6.18-rc1
References: jsc#PED-14430 jsc#PED-14297 jsc#PED-14024

Add default value for field burst_bclkdiv as initializing it to 0
is incorrect, potentially leading to a divide by 0 issue.
Valid range is 2-17 per TI datasheet. 8 was chosen as a
dummy value.

Reference <https://www.ti.com/lit/ds/symlink/tlv320dac32.pdf>.

Signed-off-by: Alex Tran <alex.t.tran@gmail.com>
Message-id: <20250901184008.1249535-1-alex.t.tran@gmail.com>
Signed-off-by: Mark Brown <broonie@kernel.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 sound/soc/codecs/tlv320dac33.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/codecs/tlv320dac33.c b/sound/soc/codecs/tlv320dac33.c
index 36b3de75a3f4..feefc777c47e 100644
--- a/sound/soc/codecs/tlv320dac33.c
+++ b/sound/soc/codecs/tlv320dac33.c
@@ -1480,6 +1480,8 @@ static int dac33_i2c_probe(struct i2c_client *client)
 
 	i2c_set_clientdata(client, dac33);
 
+	if (!dac33->burst_bclkdiv)
+		dac33->burst_bclkdiv = 8;
 	if (!dac33->mode1_latency)
 		dac33->mode1_latency = 10000; /* 10ms */
 	dac33->irq = client->irq;
-- 
2.52.0

